controllers/applicationvolumereplication_controller.go

```diff
@@ -195,295 +191,471 @@ func (r *ApplicationVolumeReplicationReconciler) Reconcile(ctx context.Context,
-			a.log.Error(err, "failed to unpause subscription", "name", subscription.Name)
+	if !reflect.DeepEqual(avrPlRule.Status.Decisions, userPlRule.Status.Decisions) {
+		avrPlRule.Status.DeepCopyInto(&userPlRule.Status)
+
+		r.Log.Info("Copied AVR PlacementRule Status into the User PlacementRule Status",
+			"Decisions", userPlRule.Status.Decisions)
 
-			return nil, requeue
+		if err := r.Status().Update(context.TODO(), userPlRule); err != nil {
+			return !updated, errorswrapper.Wrap(err, "failed to update userPlRule")
 		}
 
-		// Subscription has been unpaused. Stop processing it and wait for the next Reconciler iteration
-		a.log.Info("Subscription is unpaused. It will be processed in the next reconciler iteration",
-			"name", subscription.Name)
+		return updated, nil
+	}
 
-		return nil, requeue
+	return !updated, nil
+}
+
+type AVRInstance struct {
+	reconciler        *ApplicationVolumeReplicationReconciler
+	ctx               context.Context
+	log               logr.Logger
+	instance          *rmn.ApplicationVolumeReplication
+	needStatusUpdate  bool
+	userPlacementRule *plrv1.PlacementRule
+	avrPlacementRule  *plrv1.PlacementRule
+}
+
+func (a *AVRInstance) startProcessing() bool {
+	a.log.Info("Starting to process placement for AVR", "name", a.instance.Name)
+	requeue := a.processPlacement()
+
+	if a.needStatusUpdate {
+		if err := a.updateAVRStatus(); err != nil {
+			a.log.Error(err, "failed to update status")
+
+			requeue = true
+		}
 	}
 
-	a.log.Info("Subscription is unpaused", "name", subscription.Name)
+	a.log.Info("Completed processing placement for AVR", "name", a.instance.Name, "requeue", requeue)
+
+	return requeue
+}
+
+func (a *AVRInstance) processPlacement() bool {
+	a.log.Info("Process AVR Placement", "name", a.avrPlacementRule)
 
-	exists, err := a.vrgManifestWorkAlreadyExists(subscription)
+	requeue := true
+
+	done, err := a.runPrerequisitesForUserAction()
 	if err != nil {
-		return nil, requeue
+		a.log.Error(err, "processPlacement")
+
+		return requeue
 	}
 
-	if exists {
-		return nil, !requeue
+	if !done {
+		a.log.Info("Ran prerequisites but we'll wait until all outstanding operations complete")
+
+		return requeue
 	}
-	// This subscription is ready for manifest (VRG) creation
-	placementDecision, err := a.processUnpausedSubscription(subscription)
+
+	// Running prerequisits is complete.  That also mean that ManifestWork is applied as well.
+	// Now try to update user placementRule
+	neededUpdate, err := a.reconciler.updateUserPlacementRule(a.avrPlacementRule, a.userPlacementRule)
 	if err != nil {
-		a.log.Error(err, "Failed to process unpaused subscription", "name", subscription.Name)
+		a.log.Error(err, "processPlacement")
 
-		return nil, requeue
+		return requeue
 	}
 
-	a.log.Info(fmt.Sprintf("placementDecisions %+v - requeue: %t", placementDecision, !requeue))
+	// If we needed to update user placement rule, then we need to wait for it
+	// to have a chance to run. Hence, requeueing
+	if neededUpdate {
```

Comment:
Same comment as prior call to updateUserPlacementRule, we need not wait as we just reconciled the placementrule.

Reply:
Same answer as above. This is leftover from the subscription code.  We'll remove